app/controllers/visualizations_controller.rb

```diff
@@ -7,19 +7,16 @@ class VisualizationsController < ApplicationController
 
   # GET /visualizations.json
   def index
-    only_library = ActiveModel::Type::Boolean.new.cast(params[:onlyLibrary])
-    exclude_library = ActiveModel::Type::Boolean.new.cast(params[:excludeLibrary])
+    domain = params[:domain]
 
     # TODO: (gdingle): include samples? include projects?
-    if only_library
+    if domain == "library"
       visualizations = current_user.visualizations
-    elsif exclude_library
+    elsif domain == "public"
       visualizations = Visualization
                        .where(public_access: 1)
-    # TODO: See https://jira.czi.team/browse/IDSEQ-866
-    # .where.not(user: current_user)
     else
-      raise 'Visualizations must be for either "my library" or "public"'
+      Visualizations.where(public_access: 1, user: current_user)
```

Comment:
will this ever make sense? public visualizations by the current user but not private ones?

Reply:
I believe so, but not necessarily right now (short term for an admin option). Changed it because I prefer not raising errors unless something is strictly wrong.

I can put it back and change when necessary if you have a strong feeling about it. Wdyt?

Reply:
>  I prefer not raising errors unless something is strictly wrong

That's what I'm wondering... is there a domain other than library or public that is valid?

Reply:
Yes, the union of both (all that the user has access to). Which is our default for when we loading samples. The question would be if the same default makes sense for visualizations.

I see the library and public as filters over the default.

Reply:
Ok so wouldn't `Visualizations.where(public_access: 1, user: current_user)` be the intersection? Don't we need an `or` in there somehow?

Reply:
Yes, we are correct. That is a bug.